### DISCUSSION TITLE
useScript for CSP compatibility

### DISCUSSION THREAD

**Discussion by vejja** [Baroshem/nuxt-security#398] [General]
Hi @Baroshem

One of the most difficult thing to get working with CSP are external, 3rd-party scripts (Google Analytics, Stripe, Cloudflare Turnstile, etc.). 
We have examples in our documentation about how to get it done with `useHead` but it remains complex in real life.

With Nuxt 3.9, comes a new `useScript` utility under --experimental flag, which is absolutely fantastic and exactly what we need as a universal, ready-to-use solution. 

There is a great documentation section on `unhead` that explains very clearly why this is the right approach [here](https://unhead.unjs.io/usage/composables/use-script)

There is one small limitation though which relates to how inline event handlers are blocked by CSP. I opened up a discussion there at https://github.com/unjs/unhead/discussions/323. 

Let me know what you think and if you have additional ideas that would benefit us !

**Comment by Baroshem** [comment]
Hey @vejja 

Thank you so much for this discussion!

I do really think that having an example of usage in our docs for `useScript` would be really useful.

But if I understand correctly, we are waiting for the resolution of the discussion, am I correct?

In general, I will be more than happy to have docs about it in NuxtSecurity documentation so just waiting for green light from you :)

**Comment by vejja** [comment]
Yes you’re correct : let’s see if @harlan-zw has an idea to make CSP easier with useScript

**Comment by vejja** [comment]
Hey @Baroshem very good news:
@harlan-zw has just released a fix in `unhead@1.8.15` that allows full CSP compatibility for 3rd-party scripts 🚀🚀🚀
Let's wait for the next release of Nuxt that will incorporate this patch, and I'll propose a re-write of our docs that will make things way simpler for everybody
We are only one very small step away from allowing everyone to deploy any Nuxt application with full CSP protection, this is great !

**Reply by Baroshem** [reply]
This is indeed a great news! Thanks @vejja for that! You are a MVP ⚽️

**Comment by vejja** [comment]
Closing  - this is now part of Nuxt : see #403
